File: IUTBeans/IUTBeans/Solution-2/Condiments/Milk.cs
namespace IUTBeans.Solution_2.Condiments;

public class Milk:ICondiment
{
    private ICondiment? _condiment;
    public Milk(ICondiment? condiment)
    {
        _condiment = condiment;
    }
    public string GetDescription()
    {
        return "with Milk " + (_condiment?.GetDescription() ?? ".");
    }

    public double GetPrice()
    {
        return 50 + (_condiment?.GetPrice() ?? 0);
    }
}

File: Lab_4/src/Condiment/Milk.java
package Condiment;

public class Milk implements ICondiment {
    private ICondiment nextCondiment;

    public Milk(ICondiment nextCondiment){
        this.nextCondiment = nextCondiment;
    }
    public double getPrice(){
        return 10 + nextCondiment.getPrice();
    }
}

